### DISCUSSION TITLE
Problems running etcd tests locally

### DISCUSSION THREAD

**Discussion by nchu-stripe** [etcd-io/etcd#16404] [Q&A] [answered]
Hello, am exploring contributing to etcd. I'm having trouble with running the tests. 

I'm following the documentation provided from the CONTRIBUTING readme (https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#Setup-development-environment) 

I'm able to `make build` to actually compile etcd and run it, but the tests script seems to be broken. 

> Please ensure that your change passes tests.
> * make test-unit to run unit tests.
> * make test-integration to run integration tests.
> * make test-e2e to run e2e tests.

All of these error for me on this line: https://github.com/etcd-io/etcd/blob/main/scripts/test.sh#L690

```
for pass in $PASSES; do
  if run_pass "${pass}" "${@}"; then
    continue
```

with a "unbound variable" error. 

I tried looking among the issues and other discussion, but did not see this error. 

Note re: "Setup development environment", I have followed option 1 "Manually setup local environment.", if that matters. 

Any help would be appreciated :) thank you!

**Comment by jmhbnz** [comment]
Hey @nchu-stripe - Thanks for your question.  Can you please confirm:
- what branch of etcd you are working on, 
- which operating system you are using,
- the exact command you are running which is not working

**Reply by nchu-stripe** [reply]
Hey James! 

> what branch of etcd you are working on

have checked out main, HEAD is at 882edb3d63cca13d4b2541e4e39e29c53f3814a6

> which operating system you are using

macOS Ventura 13.4.1

> the exact command you are running which is not working

all of

make test-unit
make test-integration
make test-e2e

Thank you 🙏

**Reply by jmhbnz** [reply, marked as answer]
Hey @nchu-stripe - As mentioned in the [contributor guide](https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#setup-development-environment), the only supported development platform for the project is `linux amd64`.

Personally I don't have access to a mac so I can't look into this any further myself sorry.  Perhaps another contributor will be able to shed some light on what might be going wrong.

In the interim, as an alternative consider using a linux virtual machine for etcd development.

**Reply by jmhbnz** [reply]
Just a short follow-up, pull request https://github.com/etcd-io/etcd/pull/16433 may resolve this `unbound variable` issue once merged.
